src/widget/wtrackmenu.cpp

```diff
@@ -630,28 +623,36 @@ TrackIdList WTrackMenu::getTrackIds() const {
         const QModelIndexList indices = getTrackIndices();
         trackIds.reserve(indices.size());
         for (const auto& index : indices) {
-            trackIds.push_back(m_pTrackModel->getTrackId(index));
+            const auto trackId = m_pTrackModel->getTrackId(index);
+            if (trackId.isValid()) {
```

Comment:
The next loop uses a `DEBUG_ASSERT`. We might as well use it here.

Reply:
These are a different use cases. The TrackModel **might not** be able to successfully resolve the TrackId from QModelIndex somehow, while all Track objects that are stored in the library **must** have a valid TrackId.

Reply:
Cool. Thanks for the clarification.